cpp/nvtx3.hpp

```diff
@@ -15,21 +15,67 @@
  */
 #pragma once
 
-#if defined(NVTX3_MINOR_VERSION) and NVTX3_MINOR_VERSION < 0
-#error \
-  "Trying to #include NVTX version 3 in a source file where an older NVTX version has already been included.  If you are not directly using NVTX (the NVIDIA Tools Extension library), you are getting this error because libraries you are using have included different versions of NVTX.  Suggested solutions are: (1) reorder #includes so the newest NVTX version is included first, (2) avoid using the conflicting libraries in the same .c/.cpp file, or (3) update the library using the older NVTX version to use the newer version instead."
-#endif
-
+/* Temporary helper #defines, #undef'ed at end of header */
+#define NVTX3_CPP_VERSION_MAJOR 1
+#define NVTX3_CPP_VERSION_MINOR 0
+
+/* This section handles the decision of whether to provide unversioned symbols.
+ * If NVTX3_CPP_REQUIRE_EXPLICIT_VERSION is #defined, unversioned symbols are
```

Comment:
An example here showing good/bad file include orders would help someone understand what this is all for.

Reply:
Would you be OK with me adding that more detailed info to the DITA docs instead of here in the comments?  I think it's tricky to explain well without diagrams and pretty formatting, and I also don't like making these files get humongous just with comments.  I already have a versioning guide for NVTX and intend to do one for NVTX++ too.